src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -0,0 +1,388 @@
+    }
+
+    /**
+     * Wrapper method to check if a stack is empty.
+     * Used for easy updating to 1.11.
+     *
+     * @param stack The stack to check.
+     * @return True when the stack is empty, false when not.
+     */
+    @NotNull
+    public static Boolean isItemStackEmpty(@Nullable final ItemStack stack)
+    {
+        return stack == null || stack == EMPTY || stack.getCount() <= 0;
+    }
+
+    /**
+     * Checks if this ItemStack can be used as a Tool of type.
+     *
+     * @param itemStack Item to check.
+     * @param toolType  Type of the tool.
+     * @return true if item can be used, otherwise false.
+     */
+    public static boolean isTool(@Nullable final ItemStack itemStack, final IToolType toolType)
+    {
+        if (isItemStackEmpty(itemStack))
+        {
+            return false;
+        }
+
+        boolean isATool = false;
+        if (ToolType.AXE.equals(toolType) || ToolType.SHOVEL.equals(toolType) || ToolType.PICKAXE.equals(toolType))
+        {
+            isATool = getMiningLevel(itemStack, toolType) >= 0;
+        }
+        else if (ToolType.HOE.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemHoe;
+        }
+        else if (ToolType.BOW.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemBow;
+        }
+        else if (ToolType.SWORD.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemSword;
+        }
+        else if (ToolType.FISHINGROD.equals(toolType))
+        {
+            isATool = itemStack.getItem() instanceof ItemFishingRod;
+        }
+        else if (ToolType.HOE.equals(toolType))
+        {
+            isATool=false;
+        }
+        return isATool;
+    }
+
+
+    /**
+     * Calculate the mining level an item has as a tool of certain type.
+     *
+     * @param stack    the stack to test.
+     * @param toolType the tool category.
+     * @return integer value for mining level &gt;= 0 is okay.
+     */
+    @SuppressWarnings(DEPRECATION)
+    public static int getMiningLevel(@Nullable final ItemStack stack, @Nullable final IToolType toolType)
+    {
+        if (toolType == ToolType.NONE)
+        {
+            //empty hand is best on blocks who don't care (0 better 1)
+            return stack == null ? 0 : 1;
+        }
+        if (stack == null || stack == ItemStack.EMPTY)
+        {
+            return -1;
+        }
+        if (!Compatibility.getMiningLevelCompatibility(stack, toolType.toString()))
+        {
+            return -1;
+        }
+        //todo: use 'better' version of this thing
+        int level = -1;
+        if (ToolType.HOE.equals(toolType))
+        {
+             if (stack.getItem() instanceof ItemHoe)
+             {
+                 final ItemHoe itemHoe = (ItemHoe)stack.getItem();
+                 level = getToolLevel(itemHoe.getMaterialName());
```

Comment:
hmm. @Kostronor Imo that'd be a case where multiple returns makes sense and we can surpress that warning, what do you think?